src/ui/public/directives/paginated_selectable_list.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import paginatedSelectableListTemplate from 'ui/partials/paginated_selectable_list.html';
+
+const module = uiModules.get('kibana');
+
+module.directive('paginatedSelectableList', function (kbnUrl) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      perPage: '=',
+      list: '=',
+      userMakeUrl: '=',
+      userOnSelect: '='
+    },
+    template: paginatedSelectableListTemplate,
+    controller: function ($scope, $element, $filter) {
+      $scope.perPage = $scope.perPage || 10;
+
+      $scope.hits = $scope.list.sort() || [];
```

Comment:
Actually, re-reading this, the entire directive appear that this assume you always pass in a list of strings. I think it would be more likely that we'd have a list of objects and we'd want to pass that in.